pkg/python/python.go

```diff
@@ -0,0 +1,359 @@
+// Copyright 2022 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package python
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	apkotypes "chainguard.dev/apko/pkg/build/types"
+	"chainguard.dev/melange/pkg/build"
+	"chainguard.dev/melange/pkg/manifest"
+	"github.com/pkg/errors"
+)
+
+// PythonContext is the execution context for the python subcommand.
+type PythonContext struct {
+
+	// PackageName is the name of the python package to build and install
+	PackageName string
+
+	// PackageVersion is the version of python package to build and install
+	PackageVersion string
+
+	// PythonVersion is the version of python to build the package against
+	PythonVersion string
+
+	// PackageIndex - Client for talking to pypi
+	PackageIndex *PackageIndex
+
+	// OutDir is the output directory for the generated melange files.
+	OutDir string
+
+	// BaseURIFormat is the base URI which should contain a %s for the
+	// package name.
+	BaseURIFormat string
+
+	// AdditionalRepositories contains any additional apk repos to add
+	// to the manifest.
+	AdditionalRepositories []string
+
+	// AdditionalKeyrings contains any additional apk keys to add
+	// to the manifest.
+	AdditionalKeyrings []string
+
+	// Logger is self-explanatory
+	Logger *log.Logger
+
+	// ToGenerate is the map of dependencies that have been visited when the
+	// transitive dependency list is being calculated.
+	ToGenerate map[string]Package
+
+	//Pypi Package metadata about package
+	Package Package
+
+	// ToCheck is the list of dependencies that have yet to be checked for
+	// transitive dependencies.
+	ToCheck []string
+}
+
+// TODO: Why isn't this pulled in from melange/build?
+type ImageContents struct {
+	Repositories []string `yaml:"repositories,omitempty"`
+	Keyring      []string `yaml:"keyring,omitempty"`
+	Packages     []string `yaml:"packages,omitempty"`
+}
```

Comment:
This can likely be removed since it's in the same repository now

Reply:
its still apko_types.ImageConfiguration if im not mistaken

Reply:
Ah yes, I think that TODO is wrong. I do remember for some reason I had issues importing ImageContents from apko types but I was likely doing something wrong and settled with this workaround